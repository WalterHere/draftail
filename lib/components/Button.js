import React from 'react';

import Icon from '../components/Icon';

const onMouseDown = (onClick, e) => {
    e.preventDefault();

    onClick();
};

const Button = ({ icon, label, active, onClick }) => (
    <button
        className={`RichEditor-styleButton${active ? ' RichEditor-activeButton' : ''}`}
        type="button"
        onMouseDown={onMouseDown.bind(null, onClick)}
    >
        {icon ? (
            <Icon name={icon} />
        ) : null}
        {label}
    </button>
);

Button.propTypes = {
    label: React.PropTypes.string,
    onClick: React.PropTypes.func,
    icon: React.PropTypes.string,
    active: React.PropTypes.bool,
};

Button.defaultProps = {
    label: null,
    onClick: () => {},
    icon: null,
    active: false,
};

export default Button;
