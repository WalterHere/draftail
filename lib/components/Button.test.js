import React from 'react';
import { shallow } from 'enzyme';
import Button from '../components/Button';

describe('Button', () => {
    it('exists', () => {
        expect(Button).toBeDefined();
    });

    it('empty', () => {
        expect(shallow(<Button />)).toMatchSnapshot();
    });

    it('#label', () => {
        expect(shallow(<Button label="Test label" />)).toMatchSnapshot();
    });

    it('#icon', () => {
        expect(shallow(<Button icon="icon-test" />)).toMatchSnapshot();
    });

    it('#active', () => {
        expect(shallow(<Button active={true} />)).toMatchSnapshot();
    });

    it('#onClick', () => {
        const onClick = jest.fn();
        const event = { preventDefault: jest.fn() };
        shallow(<Button onClick={onClick} />).simulate('mousedown', event);
        expect(onClick.mock.calls.length).toBe(1);
        expect(event.preventDefault.mock.calls.length).toBe(1);
    });
});
