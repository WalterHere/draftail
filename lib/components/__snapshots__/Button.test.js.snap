exports[`Button #active 1`] = `
<button
  className="RichEditor-styleButton RichEditor-activeButton"
  onMouseDown={[Function]}
  type="button" />
`;

exports[`Button #icon 1`] = `
<button
  className="RichEditor-styleButton"
  onMouseDown={[Function]}
  type="button">
  <Icon
    className=""
    name="icon-test"
    title={null} />
</button>
`;

exports[`Button #label 1`] = `
<button
  className="RichEditor-styleButton"
  onMouseDown={[Function]}
  type="button">
  Test label
</button>
`;

exports[`Button empty 1`] = `
<button
  className="RichEditor-styleButton"
  onMouseDown={[Function]}
  type="button" />
`;
